OtripleS.Web.Api.Tests.Unit/Services/CourseAttachments/CourseAttachmentServiceTests.Exceptions.Add.cs

```diff
@@ -0,0 +1,123 @@
+﻿//---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+//----------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using OtripleS.Web.Api.Models.CourseAttachments;
+using OtripleS.Web.Api.Models.CourseAttachments.Exceptions;
+using Xunit;
+
+namespace OtripleS.Web.Api.Tests.Unit.Services.CourseAttachments
+{
+    public partial class CourseAttachmentServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnAddWhenSqlExceptionOccursAndLogItAsync()
+        {
+            // given
+            CourseAttachment someCourseAttachment = CreateRandomCourseAttachment();
+            var sqlException = GetSqlException();
+
+            var expectedCourseAttachmentDependencyException =
+                new CourseAttachmentDependencyException(sqlException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.InsertCourseAttachmentAsync(someCourseAttachment))
```

Comment:
Because we don't really care about what input parameters we pass in.
In our mocks for exceptions we should use It.IsAny<TypeName> instead of the actual input parameters we pass in

Reply:
thx for the tips. i switched it up